crates/biome_js_formatter/src/js/expressions/arrow_function_expression.rs

```diff
@@ -206,18 +212,25 @@ fn format_signature(
     arrow: &JsArrowFunctionExpression,
     is_first_or_last_call_argument: bool,
     ancestor_call_expr_or_logical_expr: bool,
+    is_first_in_chain: bool,
 ) -> impl Format<JsFormatContext> + '_ {
     format_with(move |f| {
-        if let Some(async_token) = arrow.async_token() {
-            write!(f, [async_token.format(), space()])?;
-        }
+        let formatted_async_token = format_with(|f: &mut JsFormatter| {
+            if let Some(async_token) = arrow.async_token() {
+                write!(f, [async_token.format(), space()])?;
+                Ok(())
+            } else {
+                Ok(())
+            }
+        });
 
-        let format_parameters = format_with(|f: &mut JsFormatter| {
+        let formatted_parameters = format_with(|f: &mut JsFormatter| {
             write!(f, [arrow.type_parameters().format()])?;
 
             match arrow.parameters()? {
                 AnyJsArrowFunctionParameters::AnyJsBinding(binding) => {
-                    let should_hug = is_test_call_argument(arrow.syntax())?;
+                    let should_hug =
+                        is_test_call_argument(arrow.syntax())? || is_first_or_last_call_argument;
```

Comment:
Because single parameters and lists are handled separately, the group layout condition needs to be checked here, and it was easy enough to just tack onto the `hug` condition.